velox/type/Type.h

```diff
@@ -468,6 +499,16 @@ class Type : public Tree<const std::shared_ptr<const Type>>,
 
   virtual bool isPrimitiveType() const = 0;
 
+  /// Returns unique logical type name. It can be
+  /// different from the physical type name returned by 'kindName()'.
+  virtual const char* name() const = 0;
+
+  /// Returns a possibly empty list of type parameters.
+  virtual const std::vector<TypeParameter>& parameters() const = 0;
```

Comment:
How are these parameters used during type comparison and type binding? Do we assume they all need to be the same so that two types compare equal?

Reply:
When binding decimal(10, 5) to a type signature decimal(p, s), we use these parameters to figure out p = 10 and s = 5.

Reply:
Type comparison will require the TypeParameters to be equal yes.